lambdas/service/app.py

```diff
@@ -456,21 +457,26 @@ def _fetch_integrations(entity_type: str, integration_type: str, entity_ids: Opt
     :param entity_ids: If given results will be limited to this set of entity UUIDs
     :return: A list of portal dicts that one or more matching integrations
     """
-    plugin = Plugin.load()
-    portals = plugin.portal_integrations_db()
+
     results = []
-    for portal in portals:
-        integrations = [
-            {k: v if k != 'entity_ids' else v[config.dss_deployment_stage] for k, v in integration.items()}
-            for integration in cast(Sequence[JSON], portal['integrations'])
-            if integration['entity_type'] == entity_type and integration['integration_type'] == integration_type
-        ]
-        if entity_ids is not None:
-            integrations = [integration for integration in integrations
-                            if 'entity_ids' not in integration or entity_ids.intersection(integration['entity_ids'])]
-        if len(integrations) > 0:
-            portal = {k: v if k != 'integrations' else integrations for k, v in portal.items()}
-            results.append(portal)
+
+    def list_integrations(portal_db):
```

Comment:
In the spirit of slimming down `app.py` and moving functionality into controllers and services, this functionality should probably live in the service as well. The handler function here would then essentially look like `return PortalService().list_integrations()` and the `.crud()` method could become protected (`_crud`).